bootstrap/bootstrap.go

```diff
@@ -140,14 +153,22 @@ func (b *Bootstrap) Run(ctx context.Context) (exitCode int) {
 	}
 
 	if phaseErr == nil && includePhase(`plugin`) {
-		phaseErr = b.VendoredPluginPhase()
+		phaseErr = b.VendoredPluginPhase(ctx)
 	}
 
 	if phaseErr == nil && includePhase(`command`) {
-		phaseErr = b.CommandPhase()
+		var commandErr error
+		phaseErr, commandErr = b.CommandPhase(ctx)
+		// Add command exit error info. This is distinct from a phaseErr, which is
+		// an error from the hook/job logic. These are both good to report but
+		// shouldn't override each other in reporting.
+		if commandErr != nil {
+			b.shell.Printf("user command error: %v", commandErr)
+			ext.LogError(span, commandErr)
+		}
```

Comment:
Mixed feelings about this one. I can be compelled to remove this entirely. If this stays, failures caused by the CI scripts (not the hooks, tooling, agents, etc) will be marked as errors in APM. That can be good and bad depending on what we want errors to mean for our BK agent clusters.

Reply:
Hmmm yeah it would be really nice to differentiate between a command error and a non-command error. I still think we should treat command errors are errors in APM, but can we add an ability to search non-command failures?

Reply:
Perhaps we could add a special tag to the root span for that? Something like `command_error` (vs. a phase/hook/agent error).